code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/internal/EdgePersonalizationResponseHandler.java

```diff
@@ -163,128 +199,300 @@ void fetchMessages(final List<Surface> surfaces) {
-     * found will be registered with the {@link LaunchRulesEngine}.
+     * Retrieves the previously fetched (and cached) feeds content for the provided surfaces.
      *
-     * @param propositions       A {@link List<PropositionPayload>} containing in-app message definitions
-     * @param clearExistingRules {@code boolean} if true the existing cached propositions are cleared and new message rules are replaced in the {@code LaunchRulesEngine}
-     * @param persistChanges     {@code boolean} if true the passed in {@code List<PropositionPayload>} are added to the cache
-     * @param expectedScope      {@code String} containing the app surface present in the {@code List<PropositionPayload>}
+     * @param surfaces A {@link List<Surface>} of surfaces to retrieve feeds for
+     * @return {@link Map<Surface, List<Proposition>>} containing previously fetched feeds content
      */
-    private void processPropositions(final List<PropositionPayload> propositions, final boolean clearExistingRules, final boolean persistChanges, final String expectedScope) {
-        final List<LaunchRule> parsedRules = new ArrayList<>();
+    private Map<Surface, List<Proposition>> retrievePropositions(final List<Surface> surfaces) {
+        final Map<Surface, List<Proposition>> propositionMap = new HashMap<>();
+        for (final Surface surface : surfaces) {
+            // add code-based propositions
+            final List<Proposition> propositionsList = propositions.get(surface);
+            if (!MessagingUtils.isNullOrEmpty(propositionsList)) {
+                propositionMap.put(surface, propositionsList);
+            }
+
+            if (MapUtils.isNullOrEmpty(inboundMessages)) {
+                continue;
+            }
+
+            final List<Inbound> inboundList = inboundMessages.get(surface);
+            if (MessagingUtils.isNullOrEmpty(inboundList)) {
+                continue;
+            }
+
+            final List<Proposition> inboundPropositionList = new ArrayList<>();
+            for (final Inbound message : inboundList) {
+                final PropositionInfo propositionInfo = this.propositionInfo.get(message.getUniqueId());
+                if (propositionInfo == null) {
+                    continue;
+                }
+
+                final String content = message.getContent();
+                final PropositionItem propositionItem = new PropositionItem(UUID.randomUUID().toString(), "https://ns.adobe.com/personalization/json-content-item", content);
+                final List<PropositionItem> propositionItemList = new ArrayList<>();
+                propositionItemList.add(propositionItem);
+                final Proposition proposition = new Proposition(propositionInfo.id, propositionInfo.scope, propositionInfo.scopeDetails, propositionItemList);
+                inboundPropositionList.add(proposition);
+            }
+            propositionMap.put(surface, inboundPropositionList);
```

Comment:
Here if the map already contains the surface, then you need to append contents of `inboundPropositionList` to its value. See - https://github.com/adobe/aepsdk-messaging-ios/blob/feature-feed/AEPMessaging/Sources/Messaging.swift#L475

You need a util similar to `updatePropositionMapForSurface` but it should append a list of propositions as opposed to a single proposition.